std/conv.d

```diff
@@ -306,6 +210,181 @@ template to(T)
+    int b = to!int(a);
+    double c = to!double(3.14); // c is double with value 3.14
+}
+
+/**
+ * Converting among numeric types is a safe way to cast them around.
+ *
+ * Conversions from floating-point types _to integral types allow loss of
+ * precision (the fractional part of a floating-point number). The
+ * conversion is truncating towards zero, the same way a cast would
+ * truncate. (To round a floating point value when casting _to an
+ * integral, use `roundTo`.)
+ */
+@safe pure unittest
+{
+    import std.exception : assertThrown;
+
+    int a = 420;
+    assert(to!long(a) == a);
+    assertThrown!ConvOverflowException(to!byte(a));
+
+    assert(to!int(4.2e6) == 4200000);
+    assertThrown!ConvOverflowException(to!uint(-3.14));
+    assert(to!uint(3.14) == 3);
+    assert(to!uint(3.99) == 3);
+    assert(to!int(-3.99) == -3);
+}
+
+/**
+ * When converting strings to numeric types, note that the D hexadecimal and binary
+ * literals are not handled. Neither the prefixes that indicate the base, nor the
+ * horizontal bar used to separate groups of digits are recognized. This also
+ * applies to the suffixes that indicate the type.
+ *
+ * To work around this, you can specify a radix for conversions involving numbers.
+ */
+@safe pure unittest
+{
+    auto str = to!string(42, 16);
+    assert(str == "2A");
+    auto i = to!int(str, 16);
+    assert(i == 42);
+}
+
+/**
+ * Conversions from integral types _to floating-point types always
+ * succeed, but might lose accuracy. The largest integers with a
+ * predecessor representable in floating-point format are `2^24-1` for
+ * `float`, `2^53-1` for `double`, and `2^64-1` for `real` (when
+ * `real` is 80-bit, e.g. on Intel machines).
+ */
+@safe pure unittest
+{
+    // 2^24 - 1, largest proper integer representable as float
+    int a = 16_777_215;
+    assert(to!int(to!float(a)) == a);
+    assert(to!int(to!float(-a)) == -a);
+}
+
+/**
+ * Converting an array _to another array type works by converting each
+ * element in turn. Associative arrays can be converted _to associative
+ * arrays as long as keys and values can in turn be converted.
+ */
+@safe pure unittest
+{
+    import std.string : split;
+
+    int[] a = [1, 2, 3];
+    auto b = to!(float[])(a);
+    assert(b == [1.0f, 2, 3]);
+    string str = "1 2 3 4 5 6";
+    auto numbers = to!(double[])(split(str));
+    assert(numbers == [1.0, 2, 3, 4, 5, 6]);
+    int[string] c;
+    c["a"] = 1;
+    c["b"] = 2;
+    auto d = to!(double[wstring])(c);
+    assert(d["a"w] == 1 && d["b"w] == 2);
+}
+
+/**
+ * Conversions operate transitively, meaning that they work on arrays and
+ * associative arrays of any complexity.
+ *
+ * This conversion works because `to!short` applies _to an `int`, `to!wstring`
+ * applies _to a `string`, `to!string` applies _to a `double`, and
+ * `to!(double[])` applies _to an `int[]`. The conversion might throw an
+ * exception because `to!short` might fail the range check.
+ */
+pure unittest
```

Comment:
`pure` seems to be the origin of the latest auto tester failure. Old UT for AA at line 1096 was not pure !